domain/src/main/java/org/oppia/android/domain/survey/SurveyProgressController.kt

```diff
@@ -457,6 +473,63 @@ class SurveyProgressController @Inject constructor(
     convertAsyncToAutomaticDataProvider("${baseId}_$activeSessionId")
   }
 
+  private suspend fun ControllerState.finishSurveyAndLog(isCompletion: Boolean) {
+    when {
+      isCompletion -> {
+        surveyLogger.logMandatoryResponses(
+          surveyId,
+          profileId,
+          getStoredResponse(SurveyQuestionName.USER_TYPE)!!,
+          getStoredResponse(SurveyQuestionName.MARKET_FIT)!!,
+          getStoredResponse(SurveyQuestionName.NPS)!!
+        )
+
+        // TODO(#5001): Optional responses are uploaded to Firestore, which is out of scope for this PR
```

Comment:
TODOs should state the action that should be taken to resolve the TODO. 

It is not useful to say "this PR" in it, because, when this PR gets merged into the codebase, developers won't know which PR is being referred to. Comments about scope can go in the PR description or a GitHub comment on the PR.

Reply:
Thanks, I have updated this. I have the comment in the description.